### PR TITLE
Add email support (GH-23)

### PR DESCRIPTION
**Pull request by eoehen**
- Add support to send mail after backup is executed.
- Add log volume for test or dev environment.
- Successfully tested with [PaperCut](https://github.com/ChangemakerStudios/Papercut-SMTP)

Extracted from PR #25 by @pfeigl

### CODE REVIEW

**Review by pfeigl — APPROVED** [review]
LGTM

**Comment by eoehen** [comment]
@pfeigl My tests dosn't send any emails. Do you have an idea why?

**Comment by pfeigl** [comment]
I also had some problems. What you can do is the following:
* run the mail command interactively and see if it shows any errors
* Either enable the msmtp logfile (`logfile /var/log/msmtp.log`) or install rsyslog, start the service and set `syslog on` in msmtp config.

**Comment by pfeigl** [comment]
Btw, you can use this to also get mail stdout and stderr into the cronlog 
`    echo "$cronSchedule . /container_env.sh; /usr/local/bin/backup.sh 2>&1 | tee -a /var/log/cron.log | mail -s 'SQL Server Backup Result' $MAIL_TO 2>&1 | tee -a /var/log/cron.log"`

**Comment by pfeigl** [comment]
The `-a` (append) is important, I missed them in my initial version - this would truncate the cron.log over and over again.

**Comment by eoehen** [comment]
> The `-a` (append) is important, I missed them in my initial version - this would truncate the cron.log over and over again.

Thank you for your input, it was helpful to find the problem. 
But I am not sure if we should use the append parameter (-a). I think in this case we should have the control about the growing of the logfile.

I create a new Issue for this #39

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -9,7 +67,17 @@ cronSchedule=${cronSchedule%\"}
 cronSchedule=${cronSchedule#\"}
 
 # Create crontab definition
-echo "$cronSchedule . /container_env.sh; /usr/local/bin/backup.sh >> /var/log/cron.log 2>&1" > /etc/cron.d/crontab.conf
+if [[ $SMTP_HOST ]];
+then
+    echo "Cron e-mail reporting activated. '${SMTP_HOST}'"
+    # (GH-39) Add (-a) paramter to append the log file, otherwise the log file is truncated every run.
```
Comment (eoehen) [inline docker-entrypoint.sh:73]: Added a new Issue for the append parameter. #39

**Comment by eoehen** [comment]
> @pfeigl My tests dosn't send any emails. Do you have an idea why?

Now I could test it with [PaperCut](https://github.com/ChangemakerStudios/Papercut-SMTP)
Without TLS and AUTH.

**Comment by pfeigl** [comment]
Great. I tried with mailtrap and office365 servers and both worked great (also with tls and auth).

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -9,7 +67,17 @@ cronSchedule=${cronSchedule%\"}
...
+    # echo "$cronSchedule . /container_env.sh; /usr/local/bin/backup.sh 2>&1 | tee -a /var/log/cron.log | mail -s 'SQL Server Backup Result' $MAIL_TO 2>&1 | tee -a /var/log/cron.log" > /etc/cron.d/crontab.conf
+    echo "$cronSchedule . /container_env.sh; /usr/local/bin/backup.sh 2>&1 | tee /var/log/cron.log | mail -s 'SQL Server Backup Result' $MAIL_TO 2>&1 | tee /var/log/cron.log" > /etc/cron.d/crontab.conf
```
Comment (pfeigl) [inline docker-entrypoint.sh:75]: I think the final ` > /etc/cron.d/crontab.conf` is not needed. That's what `tee` does already. It writes to stdout and forwards to the file.

**Review by georgesgoetz — APPROVED** [review]

File: docker-entrypoint.sh (hunk 2 above)
Comment (eoehen) [reply docker-entrypoint.sh:75]: IMHO this writes the cron configuration to the config file.

**Pull request merged**